lib/containers/common.pm

```diff
@@ -90,45 +90,57 @@ sub install_podman_when_needed {
     systemctl("stop firewalld") if ($host_os =~ 'tumbleweed');
 }
 
+sub install_docker_multiplatform {
+    my ($running_version, $sp, $host_os) = get_os_release;
+    if ($host_os eq 'centos') {
+        assert_script_run "dnf config-manager --add-repo=https://download.docker.com/linux/centos/docker-ce.repo";
+        # if podman installed use flag "--allowerasing" to solve conflicts
+        assert_script_run "dnf -y install docker-ce --nobest --allowerasing", timeout => 300;
+        return;
+    }
+    if ($host_os eq 'ubuntu') {
+        # Make sure you are about to install from the Docker repo instead of the default Ubuntu repo
+        assert_script_run "apt-cache policy docker-ce";
+        script_retry("apt-get -y install docker-ce", timeout => 300);
+        return;
+    }
+
+    my $pkg_name = check_var("CONTAINERS_DOCKER_FLAVOUR", "stable") ? "docker-stable" : "docker";
+    if (is_transactional) {
```

Comment:
If you move this up, then the SLE check in line 117 needs to be in an else branch. Because otherwise on a transactional SLES you would run both and that will not work.